src/functions-and-arrays.js

```diff
@@ -1,73 +1,560 @@
 // Iteration #1: Find the maximum
 
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 >= num2) {
+    return num1;
+  } else {
+    return num2;
+  }
+}
+
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
+];
+
+function findLongestWord(arr) {
+  if (arr.length === 0) {
+    return null;
+  }
+  var longest = arr[0];
+  for (var i = 0; i < arr.length; i++) {
+    if (longest.length >= arr[i].length) {
+      continue;
+    } else {
+      longest = arr[i];
+    }
+  }
+  return longest;
+}
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers(arr1) {
+  var total = 0;
+
+  arr1.forEach(element => {
+    total += element;
+  });
+  return total;
+}
+
+// Bonus
+
+const mixedArr = [6, 12, "miami", 1, true, "barca", "200", "lisboa", 8, 10];
+
+function sum(arr2) {
+  var total = 0;
+
+  for (i = 0; i < arr2.length; i++) {
+    if (typeof arr2[i] === "string") {
+      total += arr2[i].length;
+    } else if (typeof arr2[i] === "number") {
+      total += arr2[i];
+    } else if (arr2[i] === true) {
+      total++;
+    } else {
+      //HOw to throw an error?
+    }
+  }
+  return total;
+}
+
+// function sum(arr2) {
+//   var total = 0;
+
+//   arr2.forEach(element => {
+//     switch (element) {
+//       case true:
+//         total++;
+//         break;
+
+//       case typeof element === "string":
+//         total += element.length;
+//         break;
+
+//       case typeof element === "number":
+//         total += element;
+//         break;
+
+//       default:
+//         break;
+//     }
+//   });
+//   return total;
+// }
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(arr) {
+  if (arr == "") {
```

Comment:
Although this works, for best practices we would recommend to use something like ( arr.length === 0 ) to check if an array is empty. Using double equal "==" means you are using coercion and converting data types from an array to a string and it might be a little bit confusing to read and create issues in the long run.